drivers/dcf77/dcf77.c

```diff
@@ -0,0 +1,213 @@
+/*
+ * Copyright (C) 2019 HAW Hamburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+/**
+ * @ingroup     drivers_dcf77
+ * @{
+ *
+ * @file
+ * @brief       Device driver implementation for the dcf 77
+ *              longwave time signal and standard-frequency radio station
+ *
+ * @author      Michel Gerlach <michel.gerlach@haw-hamburg.de>
+ *
+ * @}
+ */
+
+#include <stdint.h>
+#include <string.h>
+
+#include "log.h"
+#include "assert.h"
+#include "xtimer.h"
+#include "periph/gpio.h"
+
+#include "dcf77.h"
+#include "dcf77_params.h"
+
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+/* Persistent level longer than 1200ms starts a new cycle  */
+#define DCF77_PULSE_START_HIGH_THRESHOLD        (1200000U)  /*~1000ms*/
+/* Every pulse send by the DCF longer than 130ms is interpreted as 1 */
+#define DCF77_PULSE_WIDTH_THRESHOLD             (140000U)   /*~130ms*/
```

Comment:
130ms vs 140ms? If this is an intended approximation, you could mention this in the comment